prow/sidecar/run.go

```diff
@@ -40,6 +43,16 @@ import (
 	"k8s.io/test-infra/prow/pod-utils/wrapper"
 )
 
+const LogFileName = "sidecar-logs-json"
+
+func LogSetup() (*os.File, error) {
+	logrusutil.ComponentInit()
+	logrus.SetLevel(logrus.DebugLevel)
+	logFile, err := ioutil.TempFile("", "sidecar-logs*.txt")
+	logrus.SetOutput(io.MultiWriter(os.Stderr, logFile))
```

Comment:
We should skip this if `err == nil`. I'm not sure what the behavior is if logFile is nil, it could cause a panic.